pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -1,65 +1,135 @@
-    spec_records = []
-    spec_level_dict = _get_single_spec_df.reference_dict.copy()
-    spec_level_dict["spectrum_id"] = spectrum.attrib["spectrumID"].split("scan=")[-1]
-
-    # Iterate children
-    for psm in spectrum.findall(".//{*}SpectrumIdentificationItem"):
-        psm_level_dict = spec_level_dict.copy()
-        psm_level_dict.update(
-            {
-                _get_single_spec_df.mapping_dict[k]: psm.attrib[k]
-                for k in _get_single_spec_df.mapping_dict
-                if k in psm.attrib
-            }
-        )
-        cv_param_info = {
-            c.attrib["name"]: c.attrib["value"] for c in psm.findall(".//{*}cvParam")
-        }
-        psm_level_dict.update(
-            {
-                _get_single_spec_df.mapping_dict[k]: cv_param_info[k]
-                for k in _get_single_spec_df.mapping_dict
-                if k in cv_param_info
-            }
-        )
+        if entry_tag == (f"{element_tag_prefix}PeptideEvidence"):
+            stage = 0
+            # Back to 0, so no cvParam gets written
+        elif entry_tag == (f"{element_tag_prefix}ModificationParams"):
+            stage = 0
+        elif entry_tag == (f"{element_tag_prefix}PeptideEvidenceRef"):
+            continue
+
+        elif stage == 1:
+            sequence, modifications, peptide_lookup = _peptide_lookup(
+                entry, entry_tag, sequence, modifications, peptide_lookup
+            )
+        elif stage == 2:
+            modification_mass_map, mod_name, fixed_mods = _get_modifications(
+                entry, entry_tag, modification_mass_map, mod_name, fixed_mods
+            )
+        elif stage == 3:
+            spec_results, spec_ident_items, spec_records = _spec_records(
+                entry,
+                entry_tag,
+                spec_results,
+                spec_ident_items,
+                spec_records,
+                mapping_dict,
+            )
+
+        elif entry_tag == (f"{element_tag_prefix}DBSequence"):
+            stage = 1
+            # Short before peptide_lookup begins
+        elif entry_tag == (f"{element_tag_prefix}AdditionalSearchParams"):
+            stage = 2
+        elif entry_tag == (f"{element_tag_prefix}Inputs"):
+            stage = 3
+            # Short before spec_record begins
+
+        elif entry_tag == (f"{element_tag_prefix}AnalysisSoftware"):
+            version = "comet_" + "_".join(
+                re.findall(r"([/d]*\d+)", entry.attrib["version"])
+            )
+
+        entry.clear()
+        if entry_tag == (f"{element_tag_prefix}SpectrumIdentificationList"):
+            return (
+                version,
+                peptide_lookup,
+                spec_records,
+                modification_mass_map,
+                fixed_mods,
+            )
 
-        spec_records.append(psm_level_dict)
-    return pd.DataFrame(spec_records)
+
+def _peptide_lookup(entry, entry_tag, sequence, modifications, peptide_lookup):
```

Comment:
ideally verbs in function name

Reply:
actually - must start with as defined in our coding standards - now valentin - you have never seen those so sorry :) :-P

Reply:
i surely will have to take another look at my docstrings...